### DISCUSSION TITLE
Show docs for item under cursor - function overloads

### DISCUSSION THREAD

**Discussion by JeviScript** [helix-editor/helix#5514] [Q&A]
Some languages support function overloading. When using  `<space> k` , it shows the documentation only for one overload variant, but I would like to be able to quickly to list though other overload options so I could quickly read what params are required. Is there a way to do so ?

**Comment by the-mikedavis** [comment]
There's a similar request for signature help (the doc popup that happens when typing function arguments): https://github.com/helix-editor/helix/issues/5403

For signature help we get a list of signatures from the language server that we can cycle through. For hover (`<space>k`) though we only get the markdown block so there isn't a way to cycle through overloads: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#hover. (There is the `MarkedString[]` value in that type but it's deprecated and I don't think it's used for this case.)

**Comment by Armando-CodeCafe** [comment]
is there a ny news on this? also frustrating me in c#

**Comment by DarkKapla** [comment]
`alt+n` and `alt+p` to cycle through overloaded functions' signatures.
